Oracle.md

```diff
@@ -0,0 +1,198 @@
+   - [Range](#range)
+      - [Example: tomorrow's temperature](#example-tomorrows-temperature)
+   - [Digit decomposition](#digit-decomposition)
+      - [Example: BTC/USD rate](#example-btcusd-rate)
+   - [Serialization and signing of outcome values](#serialization-and-signing-of-outcome-values)
+   - [Serialization of event descriptors](#serialization-of-event-descriptors)
+      - [Version 0 `enum_event_descriptor`](#version-0-enum_event_descriptor)
+      - [Version 0 `range_event_descriptor`](#version-0-range_event_descriptor)
+      - [Version 0 `digit_decomposition_event_descriptor`](#version-0-digit_decomposition_event_descriptor)
+   - [Oracle events](#oracle-events)
+      - [Version 0 `oracle_event`](#version-0-oracle_event)
+- [Oracle announcements](#oracle-announcements)
+      - [Version 0 `oracle_announcement`](#version-0-oracle_announcement)
+
+## Event descriptor
+
+An event descriptor provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of adaptors signatures for some CETs that will cover all possible outcomes of the event.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Three kinds of event outcomes are defined (note that a single event can be composed of several types).
+
+### Simple enumeration
+
+For events that have a narrow range of possible outcomes, the outcomes can simply be enumerated.
+
+#### Example: Weather tomorrow
+
+Tomorrow's weather can be represented as the set of strings `[sunny, cloudy, rainy]`.
+
+### Range
+
+When an event has numerical outcomes that cannot be easily enumerated, they can be represented as a series with:
+
+- start: the first possible outcome number
+- count: the number of possible outcomes
+- step: the increment
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base 10 exponent by which to multiply the signed number to obtain the actual outcome value.
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+count: 201
+step: 1
+unit: °C
+precision: 0
+```
+
+### Digit decomposition
+
+When a range of a numerical outcomes is large or unbounded, the oracle can represent the outcome using digit decomposition.
+
+The event descriptor should include:
+
+- base: a number representing the base in which the outcome value is decomposed and which will indicate the possible range of each digit that will be signed (e.g. 2 for binary, 10 for decimal, 16 for hex-decimal).
+- is-signed: a boolean value indicating whether the outcomes can be negative.
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base exponent by which to multiply the number represented by the composition of the digits to obtain the actual outcome value.
+- number of digits: the number of digits that the oracle will sign
```

Comment:
nit: Add note that this count does not include the sign where applicable

Reply:
Added something, let me know